index.ts

```diff
@@ -125,11 +125,18 @@ export function gretch<T = DefaultGretchResponse, A = DefaultGretchError>(
         response = (await sent).clone();
         status = response.status || 500;
 
-        if (
-          status !== 204 &&
-          parseInt(response.headers.get("Content-Length")) > 0
-        ) {
-          resolved = await response.clone()[key]();
+        if (status !== 204) {
+          if (key === "json") {
+            const text = await response.clone().text();
+            if (!text) {
+              resolved = undefined;
+            } else {
+              const parsed = JSON.parse(text);
+              resolved = parsed;
```

Comment:
```suggestion
            if (text) {
              resolved = JSON.parse(text);
            }
```

Since `resolved` is already `undefined`, can we remove the first block entirely to simplify?

Reply:
Yep good catch - writing logic branches to make the existing tests pass here.